src/jsone_encode.erl

```diff
@@ -306,6 +307,8 @@ array_values([],       Nexts, Buf, Opt) -> next(Nexts, <<(pp_newline(Buf, Nexts,
 array_values([X | Xs], Nexts, Buf, Opt) -> value(X, [{array_values, Xs} | Nexts], Buf, Opt).
 
 -spec object(jsone:json_object_members(), [next()], binary(), opt()) -> encode_result().
+object(Members, Nexts, Buf, ?OPT{canonical_form = true}=Opt) ->
+  object_members(lists:keysort(1, Members), Nexts, pp_newline(<<Buf/binary, ${>>, Nexts, 1, Opt), Opt);
```

Comment:
@pichi since it's first ordered by the first element of the tuple, it won't make any change indeed.   
I just wanted to sort by key originally and this is what I did. Is lists:sort faster in such case? If yes I will make the change.